datalad/customremotes/datalad.py

```diff
@@ -50,10 +51,17 @@ def transfer_retrieve(self, key, file):
                 return
             except Exception as exc:
                 ce = CapturedException(exc)
-                self.annex.debug("Failed to download url %s for key %s: %s"
-                                 % (url, key, ce))
-        raise RemoteError(
-            f"Failed to download from any of {len(urls)} locations")
+                cause = getattr(exc, '__cause__', None)
```

Comment:
Another suggestion (no objection to PR really):

This string formatting could actually go into `CapturedException` as an alternative (there's `format_standard` and `format_short` already. Could be `format_long`.

Reply:
True, thx! I will keep in mind for now, though. The location here is minor in the context of a special remote. In the big picture, we might want to investigate the reporting of exceptions in general (same viewpoint, but the implications for an implementation might differ). The details are not clear to me yet.